polaris-react/src/components/Banner/components/BannerExperimental/BannerExperimental.tsx

```diff
@@ -47,79 +48,72 @@ export function BannerExperimental({
 }: BannerProps) {
   const i18n = useI18n();
   const withinContentContainer = useContext(WithinContentContext);
-  const {backgroundColor, textColor, iconColor, StatusIcon} =
-    useBannerAttributes(status);
+  const isInlineIconBanner = !title && !withinContentContainer;
+  const bannerColors =
+    bannerAttributes[status][
+      withinContentContainer ? 'withinContentContainer' : 'withinPage'
+    ];
 
-  const isNoTitleBanner = !title && !withinContentContainer;
-
-  const statusIcon = (
-    <span className={styles[iconColor]}>
-      <Icon source={icon ?? StatusIcon} />
-    </span>
-  );
-
-  const closeIcon = (
-    <span className={styles[isNoTitleBanner ? 'icon-subdued' : iconColor]}>
-      <Icon source={CancelMinor} />
-    </span>
-  );
-
-  const bannerIcon = hideIcon ? null : statusIcon;
-
-  const dismissButton = onDismiss ? (
-    <Button
-      plain
-      primary
-      icon={closeIcon}
-      onClick={onDismiss}
-      accessibilityLabel={i18n.translate('Polaris.Banner.dismissButton')}
-    />
-  ) : null;
-
-  const actionButtons =
-    action || secondaryAction ? (
-      <ButtonGroup>
-        {action && (
-          <Button onClick={action.onAction} {...action}>
-            {action.content}
-          </Button>
-        )}
-        {secondaryAction && (
-          <Button onClick={secondaryAction.onAction} {...secondaryAction}>
-            {secondaryAction.content}
-          </Button>
-        )}
-      </ButtonGroup>
-    ) : null;
-
-  const bannerTitle = title ? (
-    <Text as="h2" variant="headingSm" breakWord>
-      {title}
-    </Text>
-  ) : null;
-
-  const bannerLayoutProps: BannerLayoutProps = {
-    backgroundColor,
-    textColor,
-    bannerTitle,
-    bannerIcon,
-    actionButtons,
-    dismissButton,
+  const bannerSlots: BannerLayoutProps = {
```

Comment:
Light suggestion to keep the original name `bannerLayoutProps` and/or rename to `commonBannerProps` or `sharedBannerProps` (pick your poison!)